src/torchcodec/decoders/_core/VideoDecoder.cpp

```diff
@@ -913,6 +917,11 @@ VideoDecoder::AudioFramesOutput VideoDecoder::getFramesPlayedInRangeAudio(
         (stopPts <= lastDecodedAvFrameEnd);
   }
 
+  torch::Tensor lastSamples = maybeFlushSwrBuffers();
+  if (lastSamples.numel() > 0) {
+    frames.push_back(lastSamples);
+  }
```

Comment:
Not particularly fond of the above. Maybe we *could* let `maybeFlushSwrBuffers` return a tensor of shape `(numChannels, 0)`, which could probably be `pushed_back()` unconditionally. Not sure that's better, there are probably nicer patterns I'm not seeing?

Reply:
Everything I can think of that does things unconditionally in this function is way too cute (returning a vector of tensors; using `std::copy()`). It may be more clear about intent if `maybeFlushSwrBuffers()` returns an optional so that then we don't need to use an empty tensor to indicate nothing to do.